src/Monitor.cs

```diff
@@ -233,41 +311,53 @@ public async Task StartCaptureAsync(string networkInterface, CancellationToken c
                 if (Regex.IsMatch(networkInterface, "^(?:[0-9]{1,3}\\.){3}[0-9]{1,3}$"))
                 {
                     _logger.LogDebug($"Searching for device matching {networkInterface}");
-                    _device = devices.FirstOrDefault(d => 
-                        d.Addresses != null && d.Addresses.Any(a => 
-                            a.Addr != null && a.Addr.ipAddress != null && 
-                                a.Addr.ipAddress.Equals(IPAddress.Parse(networkInterface))));
+                    _device = devices.FirstOrDefault(d =>
+                        d.Addresses != null && d.Addresses.Any(a =>
+                            a.Addr != null && a.Addr.ipAddress != null &&
+                            a.Addr.ipAddress.Equals(IPAddress.Parse(networkInterface))));
                 }
-                else 
+                else
                 {
-                    _device = devices.FirstOrDefault(x => 
+                    _device = devices.FirstOrDefault(x =>
                         x.Name.Equals(networkInterface, StringComparison.InvariantCultureIgnoreCase));
 
                     if (_device == null) // still unresolved, search by friendly name (ie. ipconfig /all)
                     {
-                        _device = devices.FirstOrDefault(x => 
-                            x.Interface.FriendlyName != null && x.Interface.FriendlyName.Equals(networkInterface, StringComparison.InvariantCultureIgnoreCase));
+                        _device = devices.FirstOrDefault(x =>
+                            x.Interface.FriendlyName != null && x.Interface.FriendlyName.Equals(networkInterface,
+                                StringComparison.InvariantCultureIgnoreCase));
                     }
                 }
+
+                _logger.LogDebug($"Starting packet capture on {GetInterfaceDisplayName((LibPcapLiveDevice)_device)} UDP:{ZWIFT_UDP_PORT}, TCP: {ZWIFT_TCP_PORT}");
+
+                deviceOpenFlags = DeviceModes.Promiscuous | DeviceModes.DataTransferUdp | DeviceModes.NoCaptureLocal;
             }
 
             if (_device == null)
             {
                 throw new ArgumentException($"Interface {networkInterface} not found");
             }
 
-            _logger.LogDebug($"Starting packet capture on {GetInterfaceDisplayName(_device)} UDP:{ZWIFT_UDP_PORT}, TCP: {ZWIFT_TCP_PORT}");
-
             // Open the device for capturing
-            _device.Open(mode: DeviceModes.Promiscuous | DeviceModes.DataTransferUdp | DeviceModes.NoCaptureLocal, read_timeout: READ_TIMEOUT);
+            _device.Open(mode: deviceOpenFlags, read_timeout: READ_TIMEOUT);
             _device.Filter = $"udp port {ZWIFT_UDP_PORT} or tcp port {ZWIFT_TCP_PORT}";
-            _device.OnPacketArrival += new PacketArrivalEventHandler(device_OnPacketArrival);
+
+            // When the companion packet decoder is provided also capture Zwift Companion packets
+            if (_companionPacketDecoder != null)
```

Comment:
I’ve implemented this on purpose because I didn’t want to break the default way the library works. 

If you think it’s fine to do both then I’ll change that.

Reply:
I've changed the Monitor so that it will always capture both Zwift and Zwift Companion messages by default.